scaffolding-ruby/lib/scaffolding.sh

```diff
@@ -0,0 +1,946 @@
+
+  scaffolding_bundle_install
+
+  # build_line "Cleaning old cached gems"
+  # _bundle clean --dry-run --force
+  # TODO fin: add cache saving of `$CACHE_PATH/vendor`
+
+  scaffolding_remove_gem_cache
+  scaffolding_fix_rubygems_shebangs
+  scaffolding_setup_app_config
+  scaffolding_setup_database_config
+}
+
+do_default_install() {
+  scaffolding_install_app
+  scaffolding_install_gems
+  scaffolding_generate_binstubs
+  scaffolding_vendor_bundler
+  scaffolding_install_libexec
+  scaffolding_fix_binstub_shebangs
+  scaffolding_run_assets_precompile
+  scaffolding_create_dir_symlinks
+  scaffolding_create_files_symlinks
+  scaffolding_create_process_bins
+}
+
+# This becomes the `do_default_build_config` implementation thanks to some
+# function "renaming" above. I know, right?
+_new_do_default_build_config() {
+  local key dir env_sh
+
+  _stock_do_default_build_config
+
+  if [[ ! -f "$PLAN_CONTEXT/hooks/init" ]]; then
+    build_line "No user-defined init hook found, generating init hook"
+    mkdir -p "$pkg_prefix/hooks"
+    cat <<EOT >> "$pkg_prefix/hooks/init"
+#!/bin/sh
+set -e
+
+export HOME="$pkg_svc_data_path"
+. '$pkg_svc_config_path/hab_env.sh'
+
+$(
+  case "$_app_type" in
+    (rails5|rails42|rails41)
+      cat <<_RAILS_
+
+# Check that the 'SECRET_KEY_BASE' environment variable is non-empty
+if [ -z "\${SECRET_KEY_BASE:-}" ]; then
+  >&2 echo "Required environment variable SECRET_KEY_BASE is not set."
+  >&2 echo "Set this package's config setting 'secret_key_base' to a"
+  >&2 echo "non-empty value. You can run 'rails secret' in development"
+  >&2 echo "to generate a random key string."
+  >&2 echo ""
+  >&2 echo "Aborting..."
+  exit 5
+fi
+_RAILS_
+      ;;
+  esac
+
+  if [[ "${_uses_pg:-}" == "true" ]]; then
+      cat <<_PG_
+
+# Confirm an initial database connection
+if ! $pkg_prefix/libexec/is_db_connected; then
+  >&2 echo ""
+  >&2 echo "A database connection is require for this app to properly boot."
+  >&2 echo "Is the database not running or are the database connection"
+  >&2 echo "credentials incorrect?"
+  >&2 echo ""
+  >&2 echo "There are 3 config setting for this package which must be set"
+  >&2 echo "correctly:"
+  >&2 echo ""
+  >&2 echo " * db.user      - The database username"
+  >&2 echo " * db.password  - The database password"
+  >&2 echo " * db.name      - The database name"
+  >&2 echo ""
+  >&2 echo "Aborting..."
+  exit 15
+fi
+_PG_
+  fi
+)
+
+# Create a directory for each app symlinked dir under $pkg_svc_var_path
+$(
+  for dir in "${scaffolding_symlinked_dirs[@]}"; do
+    echo "mkdir -pv '$pkg_svc_var_path/$dir'"
+  done
+)
+EOT
+    chmod 755 "$pkg_prefix/hooks/init"
+  fi
+
+  if [[ -f "$CACHE_PATH/default.scaffolding.toml" ]]; then
+    build_line "Appending Scaffolding defaults to $pkg_prefix/default.toml"
+    cat "$CACHE_PATH/default.scaffolding.toml" >> "$pkg_prefix/default.toml"
+  fi
+
+  env_sh="$pkg_prefix/config/hab_env.sh"
```

Comment:
Good call--I was experimenting with an idea and forgot about that name. I updated it to `app_env.sh` for now which better reflects what it contains.